### PR TITLE
Experiment: Replace style hooks coming from `@wordpress/block-editor`

### LINKED ISSUE #8071: Remove the dependency on `block-editor` from frontend blocks
### Problem statement
As described in [this PR](https://github.com/woocommerce/woocommerce-blocks/pull/7996), there are three hooks:
- `useBorderProps`
- `useColorProps`
- `useSpacingProps`
with the dependency on `@wordpress/block-editor` that has implicit dependency on `@wordpress/components`. That should be avoided on the frontend, otherwise those dependencies are externalised and loaded on the pages that use blocks with that dependency (+200kB `components.min.js` and +170kB `editor-block.min.js`).

### Solution
Remove the dependency on the hooks from the frontend code.

### Goal
Improve the performance, by decreasing the file transfer on the pages using blocks with those dependencies

pdToLP-fF-p2

### PR DESCRIPTION
**Pull request by mikejolley**
This PR Fixes #8071 by pulling in style functions from the `@wordpress/block-editor` package for use on our frontend. Style rules come from block `attributes`—we just need to translate them to CSS.

While this does duplicate methods from Gutenberg, it does:

- Remove the `@wordpress/block-editor` dependency which is unsuitable for frontend use and large
- Fix issues like https://github.com/woocommerce/woocommerce-blocks/issues/8910 due to this dependency
- Simplify implementation using a single `useStyleProps` hook

The downside is maintaining the utils inherited from Gutenberg.

While working on this and cleaning up style usage, I came across 3 blocks which had defined frontend scripts that were not used/needed. These need testing also (in instructions below). 

Fixes #8071

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Smoke test product blocks that have global styles applied.
2. Specifically test adding a Stock and a Attribute filter to the template, as well as modifying text color to ensure it still applies to the frontend.
3. Do the same for the price filter block.
4. Ensure at least one product block had custom spacing, color, and typography applied, and that change was both visible in the editor, and on the frontend after saving the template/post.

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

This will have a positive impact on the size of dependencies on the frontend.

### Changelog

> Removed the block-editor dependency on the frontend.

### CODE REVIEW

**Comment by mikejolley** [comment]
Looks like it didn't remove all dependencies in the report, but I don't see them in my local build. Will investigate, should this be the approach we want to take.

File: assets/js/base/utils/get-inline-styles.ts
Comment (gigitux) [inline assets/js/base/utils/get-inline-styles.ts:6]: Should we add `@wordpress/style-engine` in the package.json?

**Review by gigitux — COMMENTED** [review]
The code LGTM and it works!

> The downside is maintaining the utils inherited from Gutenberg.

This is my concern too. If some logic will change in the future on Gutenberg/WordPress side, we have to update the logic too :/

File: assets/js/base/utils/get-inline-styles.ts
Comment (nerrad) [inline assets/js/base/utils/get-inline-styles.ts]: Probably should avoid the use of lodash here given you've worked on extracting it everywhere?

File: assets/js/blocks/active-filters/block-wrapper.tsx
Comment (nerrad) [inline assets/js/blocks/active-filters/block-wrapper.tsx:15]: Shouldn't this be `props.attributes` (there may be more occurrences of this)?

File: assets/js/base/utils/get-inline-styles.ts
Comment (mikejolley) [reply assets/js/base/utils/get-inline-styles.ts]: Yeah I was waiting for the other to merge so I can use the new package.

File: assets/js/blocks/active-filters/block-wrapper.tsx
Comment (mikejolley) [reply assets/js/blocks/active-filters/block-wrapper.tsx:15]: It does seem so yes, good catch.

Comment (mikejolley) [reply assets/js/blocks/active-filters/block-wrapper.tsx:15]: Actually I'm not so sure. It seems like `props` are the attributes. I see several of them use:

```
<Block isEditor={ false } attributes={ parseAttributes( props ) } />
```

`parseAttributes` here uses props directory, not `props.attributes`...

**Comment by mikejolley** [comment]
@nerrad I've hesitating pushing too hard because this is work that affects mostly FSE rather than Rubik, but it would be nice to resolve. I assume next steps are to get another review from a FSE team, and raise the issue upstream to make those hooks available. Is that correct?

**Comment by nerrad** [comment]
Yes.

File: assets/js/blocks/active-filters/block-wrapper.tsx
Comment (mikejolley) [reply assets/js/blocks/active-filters/block-wrapper.tsx:15]: ~I think I see the issue; its confusing because some blocks have wrappers. I can fix this.~

I was correct initially. The blocks in this context have attributes at the top level, so using `props` here is correct. I'll update the naming to make this more obvious.

Comment (mikejolley) [reply assets/js/blocks/active-filters/block-wrapper.tsx:15]: I found some blocks (e.g. attribute filters) which don't need frontend.ts and don't need style props to function. I'm removing those because the props passed were in the wrong format.

**Comment by mikejolley** [comment]
Logged here: https://github.com/WordPress/gutenberg/issues/50334

File: assets/js/blocks/active-filters/block-wrapper.tsx
Comment (opr) [reply assets/js/blocks/active-filters/block-wrapper.tsx:15]: @mikejolley are you still planning to update the name here? There are a few other instances where `props` is used too which may be worth renaming

**Review by opr — APPROVED** [review]
I tested this and it seems to be working fine, I played with a bunch of styling options on the products block and the filter blocks, it seems global styles are working fine for all of them.

I did not do a deep code review as I am very unfamiliar with this side of the project, but I looked at all files and didn't find anything that seems obviously problematic. Maybe one of the store editing teams could do a deeper review of the actual code?

File: assets/js/blocks/active-filters/block-wrapper.tsx
Comment (mikejolley) [reply assets/js/blocks/active-filters/block-wrapper.tsx:15]: @opr No, I don't think they need renaming. `props` contain more than just attributes, and `props` is a generic term anyway.

File: assets/js/atomic/blocks/product-elements/price/block.tsx
Comment (tjcafferkey) [inline assets/js/atomic/blocks/product-elements/price/block.tsx:89]: Can't this just be `style={ styleProps.style }`?

File: assets/js/atomic/blocks/product-elements/title/block.tsx
Comment (Aljullu) [inline assets/js/atomic/blocks/product-elements/title/block.tsx]: Nit, this can be rewritten as:

```JS
				style={ isFeaturePluginBuild() ? styleProps.style : {} }
```

File: assets/js/atomic/blocks/product-elements/title/block.tsx
Comment (Aljullu) [inline assets/js/atomic/blocks/product-elements/title/block.tsx]: Same here.

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-label-block/block.tsx
Comment (Aljullu) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-label-block/block.tsx]: This could be simplified as well, if I'm not wrong:
```suggestion
			style={ styleProps.style }
```

File: assets/js/blocks/attribute-filter/block-wrapper.tsx
Comment (Aljullu) [inline assets/js/blocks/attribute-filter/block-wrapper.tsx]: Same:
```suggestion
			style={ styleProps.style }
```

File: assets/js/blocks/active-filters/block-wrapper.tsx
Comment (Aljullu) [inline assets/js/blocks/active-filters/block-wrapper.tsx]: :slightly_smiling_face: 

```suggestion
			style={ styleProps.style }
```

**Review by Aljullu — COMMENTED** [review]
This is looking great, thanks for working on this, @mikejolley!

I left some inline comments in places where we can avoid spreading objects. It's a minor change but thought I would mention anyway because it keeps code simpler.

Besides that, the only regression I found is when changing the background color of the Product Title inner block in the All Products block:

`trunk` | This branch
--- | ---
imatge | imatge

File: assets/js/atomic/blocks/product-elements/price/block.tsx
Comment (tjcafferkey) [inline assets/js/atomic/blocks/product-elements/price/block.tsx:86]:
```suggestion
			style={ style }
```

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-cart-button-block/block.tsx
Comment (tjcafferkey) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-cart-button-block/block.tsx]:
```suggestion
			style={ styleProps.style }
```

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-checkout-button-block/block.tsx
Comment (tjcafferkey) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-checkout-button-block/block.tsx]:
```suggestion
			style={ styleProps.style }
```

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-items-counter-block/block.tsx
Comment (tjcafferkey) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-items-counter-block/block.tsx:22]:
```suggestion
style={ ...styleProps.style }
```

File: assets/js/blocks/price-filter/block-wrapper.tsx
Comment (tjcafferkey) [inline assets/js/blocks/price-filter/block-wrapper.tsx]:
```suggestion
			style={ styleProps.style }
```

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-items-counter-block/block.tsx
Comment (tjcafferkey) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-items-counter-block/block.tsx:20]:
```suggestion
			style={ styleProps.style }
```

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-items-counter-block/block.tsx
Comment (tjcafferkey) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-title-items-counter-block/block.tsx]:
```suggestion
style={ styleProps.style }
```

File: assets/js/blocks/rating-filter/block-wrapper.tsx
Comment (tjcafferkey) [inline assets/js/blocks/rating-filter/block-wrapper.tsx]:
```suggestion
			style={ styleProps.style }
```

File: assets/js/blocks/stock-filter/block-wrapper.tsx
Comment (tjcafferkey) [inline assets/js/blocks/stock-filter/block-wrapper.tsx]:
```suggestion
			style={ styleProps.style }
```

**Review by tjcafferkey — COMMENTED** [review]
Looks good and tests well @mikejolley ! I wasn't able to find any regressions that haven't been highlighted by others.

Regarding the code, I saw @Aljullu 's comment:

>I left some inline comments in places where we can avoid spreading objects. It's a minor change but thought I would mention anyway because it keeps code simpler.

I also left a few minor inline comments for opportunities to do this, for consistency more than anything else.

**Comment by mikejolley** [comment]
Thanks for the reviews @opr @Aljullu @tjcafferkey — I've adjusted those spreads, fixed the missing styles in atomic blocks, and retested. Tests are green so merging!

**Pull request merged**
